applications/matter_weather_station/README.rst

```diff
@@ -156,6 +116,48 @@ Configuration
 
 |config|
 
+.. _matter_weather_station_app_build_types:
+
+Matter weather station build types
+==================================
+
+The Matter weather station application does not use a single :file:`prj.conf` file.
+Configuration files are provided for different build types, and they are located in the :file:`configuration/thingy53_nrf5340_cpuapp` directory.
+Before you start testing the application, you can select one of the build types supported by the application, depending on the building method.
+
+See :ref:`app_build_additions_build_types` and :ref:`modifying_build_types` for more information about this feature of the |NCS|.
+
+The application supports the following build types:
+
+.. list-table:: Matter weather station build types
+   :widths: auto
+   :header-rows: 1
+
+   * - Build type
+     - File name
+     - Supported board
+     - Description
+   * - Debug
+     - :file:`prj.conf`
+     - All from `Requirements`_
+     - | Debug version of the application; can be used to enable additional features for verifying the application behavior, such as logs or command-line shell.
+       | Used by default if no build type is explicitly selected.
+   * - Release
+     - :file:`prj_release.conf`
+     - All from `Requirements`_
+     - | Release version of the application; can be used to enable only the necessary application functionalities to optimize its performance.
+       |
+       | .. note::
+       |     Currently, this application supports only the ``release`` build type when `Building for the nRF7002 Wi-Fi expansion board`_.
+   * - Factory data
+     - :file:`prj_factory_data` under :file:`child_image/multiprotocol_rpmsg`
```

Comment:
I think something is wrong in this readme. I would say factory_data configuration should not be called a build type, as it is not enabled using any prj_factory_data.conf. The `prj_factory_data.conf` you mentioned is not related to the application, as the other ones above. It is used only for the network core image. And to be honest I think it may not be currently used anywhere (it must be some leftover after refactor). Currently we enable factory data using `overlay-factory_data.conf`.

Reply:
I've removed mentions of the build type. Rest will be handled in https://nordicsemi.atlassian.net/browse/KRKNWK-18203, as agreed.